doc/source/user_guide/categorical.rst

```diff
@@ -622,9 +622,10 @@ even if some categories are not present in the data:
     s = pd.Series(pd.Categorical(["a", "b", "c", "c"], categories=["c", "a", "b", "d"]))
     s.value_counts()
 
-``DataFrame`` methods like :meth:`DataFrame.sum` also show "unused" categories.
+``DataFrame`` methods like :meth:`DataFrame.sum` also show "unused" categories:
 
 .. ipython:: python
+   :okwarning:
```

Comment:
same, generally we want to fix any docs (except older what'snew where putting okwarning is ok)

Reply:
Sure, but to be clear on this one, fixing this requires hard-coding observed=False (or True) [here](https://github.com/pandas-dev/pandas/blob/master/pandas/core/generic.py#L10548), which someone will eventually not love. Likely going to need to add another keyword, especially, when you change the default behavior down the line.

Reply:
if it does that's a problem
we should be able to hard code in the docs not the code

if there is some lower level where the warning is coming from then that is. problem and needs to be addressed

this is why we need to NOT just suppress warnings in tests as they could be hising a bug

Reply:
There was one other place I had to do this. Any code path that uses a groupby internally is going to raise this warning with a categorical. Hopefully the test coverage is high enough not to make things super noisy for a lot of folks...

Reply:
> if it does that's a problem

Every place you use groupby internally but don't allow passing all the keywords down that control it's type-dependent behavior, this can come up. If the code path is covered in tests then the warnings-as-errors should catch it. If not, then you may hear from some folks.